include/dlaf/multiplication/general.h

```diff
@@ -20,11 +20,45 @@
 #include <dlaf/matrix/distribution.h>
 #include <dlaf/matrix/index.h>
 #include <dlaf/matrix/matrix.h>
+#include <dlaf/matrix/matrix_ref.h>
 #include <dlaf/multiplication/general/api.h>
 #include <dlaf/util_matrix.h>
 
 namespace dlaf::multiplication::internal {
 
+/// General sub-matrix distributed multiplication, computing
+/// C = alpha * A * B + beta * C
+///
+/// @param  mat_a contains the input matrix A.
+/// @pre @p mat_a is distributed according to @p grid
+///
+/// @param  mat_b contains the input matrix B.
+/// @pre @p mat_b is distributed according to @p grid
+///
+/// @param  mat_c On entry it contains the input matrix C. On exit matrix tiles in the range will be
+///         overwritten with the result, while others are left untouched.
+/// @pre @p mat_c is distributed according to @p grid
+///
+/// @pre multipliable_sizes(mat_a.size(), mat_b.size(), mat_c.size(), opA, opB)
+/// @pre multipliable_sizes(mat_a.tile_size(), mat_b.tile_size(), mat_c.tile_size(), opA, opB)
+/// @pre multipliable_sizes(mat_a.tile_size_of({0, 0}), mat_b.tile_size_of({0, 0}),
+///      mat_c.tile_size_of({0, 0}), opA, opB)
```

Comment:
Some doubts about how to expose pre-conditions:
- high level `multipliable` vs more verbose `multipliable_sizes`
- how to expose rank-alignment requirement (currently just NN is implemented, this requirement might likely be op dependent)
- should we expose here also the constraint imposed by panel (i.e. blocksize == tilesize).
  On one side it is a good idea to document, so that the user knows without inspecting code (after getting an assert triggered), on the other hand it would easily become difficult to keep track of all the places where a constraint like this is involved (e.g. it gets fixed/improved and you have to update all docs, it might not be so straightforward to do for many reasons)

Reply:
Ok, so I will leave it like this for now.